### PR TITLE
chore (runtask) : Link get pod support in runtask

### PR DESCRIPTION
**Pull request by imazik**
This PR links get pod support in runtask and fix Job get action in runtask.

Signed-off-by: Shovan Maity <shovan.maity@mayadata.io>

### CODE REVIEW

File: pkg/client/k8s/k8s.go [hunk 1]
```diff
@@ -285,10 +285,11 @@ func (k *K8sClient) GetStorageV1SCAsRaw(name string) (result []byte, err error)
 }
 
 // GetBatchV1JobAsRaw returns a Job instance
-func (k *K8sClient) GetBatchV1JobAsRaw(name string) (result []byte, err error) {
+func (k *K8sClient) GetBatchV1JobAsRaw(name, namespace string) (result []byte, err error) {
 	return k.cs.BatchV1().RESTClient().
 		Get().
 		Resource("jobs").
+		Namespace(namespace).
```
Comment (AmitKumarDas) [inline pkg/client/k8s/k8s.go]: I think k already has namespace field. Can we use that instead?

File: pkg/task/task.go [hunk 2]
```diff
@@ -1071,14 +1074,27 @@ func (m *taskExecutor) getCoreV1PV() (err error) {
 
 // getBatchV1Job will get the Job as specified in the RunTask
 func (m *taskExecutor) getBatchV1Job() (err error) {
-	job, err := m.getK8sClient().GetBatchV1JobAsRaw(m.getTaskObjectName())
+	job, err := m.getK8sClient().GetBatchV1JobAsRaw(m.getTaskObjectName(), m.getTaskRunNamespace())
```
Comment (AmitKumarDas) [inline pkg/task/task.go]: is this required? check my earlier comments

File: pkg/client/k8s/k8s.go (hunk 1 above)
Comment (imazik) [reply pkg/client/k8s/k8s.go]: Yes we can use that. Updated.

File: pkg/task/task.go [hunk 3]
```diff
@@ -1079,6 +1082,19 @@ func (m *taskExecutor) getBatchV1Job() (err error) {
 	return
 }
 
+// getV1Pod will get the Pod as specified in the RunTask
+func (m *taskExecutor) getV1Pod() (err error) {
```
Comment (sagarkrsd) [inline pkg/task/task.go]: Can we rename this as `getCoreV1Pod` .

File: pkg/task/task.go [hunk 4]
```diff
@@ -1079,6 +1082,19 @@ func (m *taskExecutor) getBatchV1Job() (err error) {
...
+	podClient := pod.NewKubeClient(pod.WithNamespace(m.getTaskRunNamespace()))
+
```
Comment (sagarkrsd) [inline pkg/task/task.go]: space not required

File: pkg/task/task.go [hunk 5]
```diff
@@ -1079,6 +1082,19 @@ func (m *taskExecutor) getBatchV1Job() (err error) {
...
+	pod, err := podClient.GetRaw(m.getTaskObjectName(), metav1.GetOptions{})
+	if err != nil {
+		return
+	}
+
```
Comment (sagarkrsd) [inline pkg/task/task.go]: same here

File: pkg/task/task.go (hunk 3 above)
Comment (imazik) [reply pkg/task/task.go]: done

**Pull request merged**